controller/state/statedb.go

```diff
@@ -742,14 +742,37 @@ func (s *stateDB) PublishRecordsFrom(ctx context.Context, worker string) error {
 			if err != nil {
 				return err
 			}
-			sig, err := s.PrivKey.Sign([]byte(lnk))
+			c, err := cid.Decode(lnk)
 			if err != nil {
 				return err
 			}
-			c, err := cid.Decode(lnk)
+
+			// check if we should include it.
+			rcrdRdr, err := txContextLoader(ctx, tx)(cidlink.Link{c}, ipld.LinkContext{})
+			if err != nil {
+				return err
+			}
+			tskBuilder := tasks.Type.FinishedTask.NewBuilder()
```

Comment:
you might want the repr prototype :)

Reply:
the non-repr version can be used for decoding, right? I've only run into the repr issue on the encoding side... have i done something wrong? 😱

Reply:
As far as I'm aware, if you use the repr node for encoding, you should use the repr prototype for decoding. Otherwise it's a mismatch. A struct with stringjoin repr encoded as repr will result in a string, but decoded into the non-repr struct will fail, for example.